app/scripts/lib/local-store.js

```diff
@@ -63,15 +63,28 @@ export default class ExtensionStore {
     if (!this.isSupported) {
       return undefined;
     }
-    const result = await this._get();
-    // extension.storage.local always returns an obj
-    // if the object is empty, treat it as undefined
-    if (isEmpty(result)) {
-      this.mostRecentRetrievedState = null;
+    // State can become corrupted on firefox which seems to be most commonly
+    // caused by a broken link between the sqlite entry for the extension and
+    // the name of the file that stores the data. Some users have been able to
+    // restore the functionality by finding and fixing the reference but until
+    // then the extension presents as an infinite spinner loader. To avoid this
+    // we can return undefined if we get an error loading state. This will make
+    // Firefox behave the same way that Chrome does when state is missing which
+    // is to load the default state and drop into the onboarding flow.
+    try {
+      const result = await this._get();
+      // extension.storage.local always returns an obj
+      // if the object is empty, treat it as undefined
+      if (isEmpty(result)) {
+        this.mostRecentRetrievedState = null;
+        return undefined;
+      }
+      this.mostRecentRetrievedState = result;
+      return result;
+    } catch (err) {
+      log.error('error getting state from local store:', err);
       return undefined;
```

Comment:
I think we may want to return `{ }` here instead of `undefined`, because in background.js we check `versionedData && !versionedData.data` to determine we should restore, which wouldn't work for `undefined`.